src/attack.rs

```diff
@@ -95,17 +97,23 @@ fn player_attack(
 }
 
 pub fn enemy_attack(
-    mut query: Query<&mut State, (With<Enemy>, With<Target>)>,
+    mut query: Query<(Entity, &mut State, &Handle<FighterMeta>), (With<Enemy>, With<Target>)>,
     mut event_reader: EventReader<ArrivedEvent>,
     mut commands: Commands,
+    fighter_assets: Res<Assets<FighterMeta>>,
 ) {
     for event in event_reader.iter() {
-        if let Ok(mut state) = query.get_mut(event.0) {
+        if let Ok((entity, mut state, fighter_handle)) = query.get_mut(event.0) {
+            let fighter = fighter_assets
+                .get(fighter_handle)
+                .expect("Fighter asset not found");
```

Comment:
I think it is possible that the fighter asset isn't loaded at this point, though I'll admit it's unlikely that the user manages to press the attack button before the figher asset finishes loading.

Still, I think we should probably put this in an `if let Some(fighter) = fighter_assets.get(fighter_handle {}`, or if it's more readable, to avoid indenting everything we could do this:

```rust
let fighter = if let Some(fighter) = fighter_assets.get(fighter_handle) {
    fighter
} else {
    // Skip this iteration of the loop
    continue;
};
```